snow/engine/snowman/engine_test.go

```diff
@@ -3182,3 +3185,35 @@ func TestShouldIssueBlock(t *testing.T) {
 		})
 	}
 }
+
+type mockConnVDR struct {
+	tracker.Peers
+	percent float64
+}
+
+func (m *mockConnVDR) ConnectedPercent() float64 {
+	return m.percent
+}
+
+func TestEngineAbortQueryWhenInPartition(t *testing.T) {
+	require := require.New(t)
+
+	// Buffer to record the log entries
+	buff := gbytes.NewBuffer()
+
+	conf := DefaultConfig(t)
+	// Overwrite the log to record what it says
+	conf.Ctx.Log = logging.NewLogger("", logging.NewWrappedCore(logging.Info, buff, logging.Plain.ConsoleEncoder()))
+	conf.Params = snowball.DefaultParameters
+	conf.ConnectedValidators = &mockConnVDR{percent: 0.7, Peers: conf.ConnectedValidators}
+
+	_, _, _, _, engine := setup(t, conf)
+
+	// Gossip will cause a pull query if enough stake is connected
+	require.NoError(engine.Gossip(context.Background()))
+
+	shouldSay := gbytes.Say("dropped query for block {\"reason\": \"insufficient stake connected\", \"blkID\": \"j7sgHpXKn6y22DQdBNr5bzidt345dor9L8rL3Mxo9bt91k4ta\", \"ratio\": 0.7}")
```

Comment:
Could we go one step further to make sure that this test passes as long as nothing has broken? For example, could we create an error type that will be included in the log when this condition is triggered and then match the logs on only that string? I like to avoid the case where changing an arbitrary string like a log message or the string inside of an error will cause a test to fail.

Reply:
Fair point, done.